src/service/action-impl.js

```diff
@@ -271,9 +273,17 @@ export class ActionService {
       return;
     }
 
+    // Check if this is an element has a known action.
+    const supportedActions = ELEMENTS_ACTIONS_MAP_[lowerTagName];
+    if (target[ACTION_QUEUE_] && supportedActions &&
```

Comment:
Not quite. We indeed initialize to an empty array. You can see it just 5 lines above for an `amp-` case (I mentioned that this should probably be the same code). What this gives you is that it buffers up actions in an array until `installActionHandler` is called. As you can see [here](https://github.com/ampproject/amphtml/blob/master/src/service/action-impl.js#L183), when `installActionHandler` is called, it goes over buffered actions and sends them to the handler. This way we remove risk condition between when a user can activate an action and when `installActionHandler` is called. That makes sense since `installActionHandler` installation depends on asynchronous script download.

In cases like this it's possible that it's "too late" to execute an action. If this happens, we can add "deferred" flag to an action that's not executed immediately. We do this [here](https://github.com/ampproject/amphtml/blob/master/src/custom-element.js#L1207), but to my knowledge no actions use it yet. So we can punt on this here.